code/modules/scavstation_shared_content/items/egg_yinglet.dm

```diff
@@ -0,0 +1,86 @@
+//yinglet nests and yinglet eggs
+
+//nest
+/obj/structure/yinglet_nest
+	name = "eggbed"
+	icon = 'icons/egg_yinglet.dmi'
+	icon_state = "base"
+	w_class = ITEM_SIZE_HUGE
+	var/has_egg = TRUE
+
+/obj/structure/yinglet_nest/empty
+	has_egg = FALSE
+
+/obj/structure/yinglet_nest/Initialize()
+	. = ..()
+	if(has_egg)
+		new /obj/item/yinglet_egg(src)
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/proc/update_desc()
+	desc = "A pile of plain but very soft fabric arranged to carefully hold an egg, as well as prevent the newly hatched younglet from smashing their face when they flop out."
+	if(has_egg && contents.len)
+		for(var/obj/item/I in contents)
+			desc += " There's \a [I.name] sitting in it."
+
+/obj/structure/yinglet_nest/attack_hand(mob/user)
+	. = ..()
+	if(has_egg && user.a_intent == I_HELP)
+		for(var/obj/item/I in contents)
+			src.visible_message(SPAN_NOTICE("[user.name] pets \the [I.name]. Grow up big and strong now, okay?"))
+			user.setClickCooldown(1.5 SECONDS)
+		return
+	if(has_egg && user.a_intent != I_HELP)
+		remove_egg(user)
+		return
+
+/obj/structure/yinglet_nest/attackby(obj/item/object, mob/user)
+	. = ..()
+	if(object.w_class > ITEM_SIZE_NORMAL || istype(object, /obj/item/holder))
+		to_chat(user, SPAN_WARNING("The [object.name] is too big for the [name]!"))
+		return
+	else if(has_egg && contents.len)
+		for(var/obj/item/I in contents)
+			to_chat(user, SPAN_WARNING("The [name] already has \a [I.name] in it!"))
+		return
+	else if(!has_egg && object.w_class <= ITEM_SIZE_NORMAL)
+		add_egg(object, user)
+		return
+
+/obj/structure/yinglet_nest/proc/remove_egg(mob/user)
+	has_egg = FALSE
+	for(var/obj/item/E in contents)
+		user.put_in_hands(E)
+		src.visible_message(SPAN_WARNING("[user.name] takes \the [E.name] from the eggbed."))
```

Comment:
`\The [user]` and `\the [E]` would work better. As-is, this won't be capitalized if a monkey or other mob grabs the egg rather than a named human.

Reply:
Gotcha. I assume I need to do this for all the visible_messages.

Reply:
Probably wise yeah.

Reply:
Also, should i just omit `x.name` for most cases?

Reply:
`[thing]` is generally preferred as far as I know as it will automatically handle `\the` if you forget or omit it.

Reply:
Gotcha.